### PR TITLE
Security level change is now handled by signals

### LINKED ISSUE #6757: Security Level Signal Handler
Security level should have a global signal, instead of looping through all airlocks, airlocks that unlock on red alert should register that signal.

### PR DESCRIPTION
**Pull request by Dejaku51**
## About The Pull Request
Makes it so security alert changes are handled by signal instead of just looping multiple times
Also changes set_security_level() code so it has early return instead of entire code being in if statement

Closes #6757 
## Why It's Good For The Game
Requested by @PowerfulBacon 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

1. Change security level to blue
2. Fire alarm light turns blue
3. Change security level to green
4. Fire alarm light turns green

## Changelog
:cl:
code: set_security_level() now uses signals instead of multiple loops
/:cl:

### CODE REVIEW

File: code/modules/shuttle/emergency.dm [hunk 1]
```diff
@@ -613,6 +613,15 @@
 	density = FALSE
 	clockwork = TRUE //it'd look weird
 
+/obj/machinery/computer/shuttle_flight/pod/Initialize()
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_SECURITY_ALERT_CHANGE, .proc/handle_alert)
+
+/obj/machinery/computer/shuttle_flight/pod/proc/handle_alert(datum/source, new_alert)
+	SIGNAL_HANDLER
+	admin_controlled = (new_alert > SEC_LEVEL_RED)
```
Comment (PowerfulBacon) [inline code/modules/shuttle/emergency.dm]: Is this right?

File: code/game/machinery/doors/door.dm [hunk 2]
```diff
@@ -68,6 +68,15 @@
 	//doors only block while dense though so we have to use the proc
 	real_explosion_block = explosion_block
 	explosion_block = EXPLOSION_BLOCK_PROC
+	if(red_alert_access)
+		RegisterSignal(SSdcs, COMSIG_GLOB_SECURITY_ALERT_CHANGE, .proc/handle_alert)
+
+/obj/machinery/door/proc/handle_alert(datum/source, new_alert)
+	SIGNAL_HANDLER
+	if(new_alert <= SEC_LEVEL_RED )
```
Comment (PowerfulBacon) [inline code/game/machinery/doors/door.dm]: Is this correct?

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/modules/shuttle/emergency.dm (hunk 1 above)
Comment (Dejaku51) [reply code/modules/shuttle/emergency.dm]: wait... fuck

**Comment by Dejaku51** [comment]
i hate myself

File: code/game/machinery/doors/door.dm (hunk 2 above)
Comment (Dejaku51) [reply code/game/machinery/doors/door.dm]: should be good now

File: code/modules/shuttle/emergency.dm (hunk 1 above)
Comment (Dejaku51) [reply code/modules/shuttle/emergency.dm]: should be good now

**Review by PowerfulBacon — APPROVED** [review]
Looks good :+1:

**Pull request merged**